packages/components/pie-modal/src/index.ts

```diff
@@ -342,6 +343,7 @@ export class PieModal extends RtlMixin(LitElement) implements ModalProps {
             <div class="c-modal-contentInner">
                 <slot></slot>
             </div>
+            ${this.isLoading ? html`<pie-spinner size="xl" variant="secondary"></pie-spinner>` : nothing}
```

Comment:
Tests?

Reply:
we already have both visual and component tests in place, this PR just refactors the implementation of the behaviour.